aws-android-sdk-auth-core/src/main/java/com/amazonaws/mobile/auth/core/IdentityManager.java

```diff
@@ -223,7 +223,9 @@ public IdentityManager(final Context context,
                            final AWSConfiguration awsConfiguration) {
         this.appContext = context.getApplicationContext();
         this.awsConfiguration = awsConfiguration;
-        this.clientConfiguration = new ClientConfiguration().withUserAgent(awsConfiguration.getUserAgent());
+        this.clientConfiguration = new ClientConfiguration()
+                .withUserAgent(awsConfiguration.getUserAgent())
+                .withUserAgentOverride(awsConfiguration.getUserAgentOverride());
```

Comment:
Do we need both? Why not just use `withUserAgent` and have it set a single variable? If not provided, that variable could default to its current `DEFAULT_USER_AGENT` (or whatever.)

If we have two, there are four states I need to consider:
1. What is the behavior when neither is provided?
2. What is the behavior when only the user agent is provided?
3. Versus only the override?
4. Or what if both are provided?

What if either are null?

Reply:
We do need both because user agent is simply appended on to default values whereas when userAgentOverride is used it changes the behavior (as you can see in this PR) to overwrite the entire value with whatever is in there.

1) A default value is used
2) User agent is appended on to the default value
3) The override is used as the entire value (without the default)
4) Same as 3

No problem if either are null.

Reply:
@TrekSoft I'm a bit confused by the use-case 2. Why is it necessary to ever append another user agent rather than always overriding?

Reply:
This is the current default behavior that AWSMobileClient is doing - don't know why all they designed it that way but here I'm simply leaving it working the way it always did before but with the option for Amplify to add this override so it replaces everything with the value we pass in.

Reply:
All of the other SDKs in `aws-sdk-android` use the following mechanism to override the `User-Agent` header:
```java
ClientConfiguration clientConfiguration = new ClientConfiguration();
clientConfiguration.setUserAgent(UserAgent.string());
```

I would prefer if we could do the exact same, or something very similar, in the `AWSMobileClient`.

Reply:
The problem is, if we just set user agent, the AWS Mobile Client value gets appended to it: https://github.com/aws-amplify/aws-sdk-android/blob/3625c63248e043fb6173f2f09d611a49cd7c6c92/aws-android-sdk-mobile-client/src/main/java/com/amazonaws/mobile/client/internal/oauth2/OAuth2Client.java#L503

Let's call and talk about this.

Reply:
That's probably okay, isn't it?

Reply:
(We talked offline, @TrekSoft to list some requirements in PR description.)

Reply:
Done